tensordict/tensordict.py

```diff
@@ -2140,7 +2147,9 @@ def is_locked(self, value: bool):
 
     def lock(self):
         self._is_locked = True
-        for key in self.keys():
+        keys_view = _TensorDictKeysView(tensordict=self, include_nested=True, leaves_only=False,
+                                        error_on_loop=False)
+        for key in keys_view:
```

Comment:
The idea with `include_nested=True` would be to use that instead of recursion. In this case we call `.lock` on all nested tensordicts, which will result in other keys_views being created and we will iterate over all the contents multiple times.

I think we should change

```diff
- self.get(key).lock()
+ self.get(key)._is_locked = True
```

To avoid the recursive calls.

Reply:
Corrected